### ISSUE TITLE
Install location not CIS compliant

### ISSUE THREAD

**Issue by Fabian1976** [wazuh/wazuh#15209]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| all | all | Manager/Agent | Packages | RHEL |

The Wazuh/OSSec components install themselves in `/var/ossec`. To comply to the CIS benchmark, a `noexec` mount options should be applied to the `/var` partition. But when we do this, all ossec components stop working because all executables are installed on the /var partition.

Could you change this behavior to comply more to most Linux programs? So install the software in `/opt` of `/usr`, logs in `/var/log` and configs in `/etc`?

**Comment by sempervictus** [comment]
Everything is relative, so to speak, because OSSEC is laid out that way to permit components to run in a chroot of that subtree isolating them from the host system somewhat. So while from the perspective of the mount context in which the CIS checks run that's `/var/ossec`, from the perspective of some of the daemons its actually `/`.

**Comment by Fabian1976** [comment]
I understand, but that is not entirely accurate i think. It is able to scan the real `/` which is 2 levels up from its root.

But still. I allways found it weird that it is installed in `/var`. I hope you can move it to comply to regular standards. I know RHEL8 is not officially supported yet, but we are using it on RHEL8 and a new CIS rule in RHEL8 is to have `noexec` on `/var`. For now we disabled that rule to be able to run it.

**Comment by vikman90** [comment]
Hi everyone,

Thank you all for your comments. This issue is related to #9175, in which we have proposed a new directory layout. The idea is to allow installing the agent anywhere in the system, but the default base directory would be `/usr/share`.

On the other hand, we will remove the `chroot` feature from all components that still use it.

We would love to receive feedback on the directory layout issue.

Best regards.

**Comment by sempervictus** [comment]
@vikman90: given that any implementation of the HIDS pipeline is still processing untrusted inputs from varying sources, i think that removing segmentation/defensive mechanisms from the stack is not the greatest of ideas - most folks aren't running grsec, preloading hardened malloc, and wrapping it in various namespacing jails to approximate the effect of the `chroot`. It will also create havoc for any scripts relying on the decades of known paths. There's going to be a bunch of churn and crying when the new paradigm lands anyway, but users learn and adapt - if a security problem gets introduced at the same time, that could shake confidence if coupled with a relaxation of controls.

**Comment by vikman90** [comment]
Hi @sempervictus,

Thanks for your comment, I really wanted to get feedback on this. Paradoxically, _chroot_ is only applied where it's least needed, IMO.

It has been never implemented in the components that need to access external resources:
- Logcollector (log reading and command execution).
- FIM (file reading and pre-filter command execution).
- Active Response (command execution). I think this is the most critical component as AR are requested by the manager.
- Rootcheck (system inspection).
- Modulesd (most modules need to run commands).

The only daemon where we have seen the need to remove _chroot_ was Agentd as that prevented it from resolving hostnames (#2770). Now, only these daemons run _chroot_:

- Analysisd (log decoding and rule matching).
- DBD (alert exporting to an external database).
- Monitord (agent state monitorization and log rotation).
- Maild (alert exporting to e-mails).
- Remoted (agent communication).
- Wazuh DB (internal database management).

In fact, these components have time zone issues (#4365), because the installer needs to copy _/etc/localtime_ into _/var/ossec/etc/localtime_.

In summary, we see that chroot gives more problems than benefits. This is why we plan to disable it. On the other hand, we are redoubling our efforts to avoid security flaws such as path traversal or shell injection (#15052).

However, if you know of reasons why this feature might be necessary, we'd be happy to discuss it.

Best regards.

**Comment by ochurlaud** [comment]
Hi, 
do we know where (and when) this issue is in the Wazuh roadmap? We need to become CIS compliant, and building the RPM ourselves is not an easy option in our enterprise context.  Thank you

**Comment by vikman90** [comment]
@ochurlaud this task is still in the icebox. Maybe we'll schedule it soon.

We'll keep you posted.

Best regards.
